src/tools/illink/src/linker/Linker/MemberActionStore.cs

```diff
@@ -49,10 +54,69 @@ public bool TryGetMethodStubValue (MethodDefinition method, out object? value)
 			if (PrimarySubstitutionInfo.MethodStubValues.TryGetValue (method, out value))
 				return true;
 
-			if (!TryGetSubstitutionInfo (method, out var embeddedXml))
+			if (TryGetSubstitutionInfo (method, out var embeddedXml)
+				&& embeddedXml.MethodStubValues.TryGetValue (method, out value))
+				return true;
+
+			if (TryGetFeatureCheckValue (method, out bool bValue)) {
+				value = bValue ? 1 : 0;
+				return true;
+			}
+
+			return false;
+		}
+
+		internal bool TryGetFeatureCheckValue (MethodDefinition method, out bool value)
+		{
+			value = false;
+
+			if (!method.IsStatic)
+				return false;
+
+			if (method.ReturnType.MetadataType != MetadataType.Boolean)
+				return false;
+
+			if (FindProperty (method) is not PropertyDefinition property)
 				return false;
 
-			return embeddedXml.MethodStubValues.TryGetValue (method, out value);
+			if (property.SetMethod != null)
+				return false;
+
+			foreach (var featureSwitchDefinitionAttribute in _context.CustomAttributes.GetCustomAttributes (property, "System.Diagnostics.CodeAnalysis", "FeatureSwitchDefinitionAttribute")) {
+				if (featureSwitchDefinitionAttribute.ConstructorArguments is not [CustomAttributeArgument { Value: string switchName }])
+					continue;
+
+				// If there's a FeatureSwitchDefinition, don't continue looking for FeatureGuard.
+				// We don't want to infer feature switch settings from FeatureGuard.
+				return _context.FeatureSettings.TryGetValue (switchName, out value);
```

Comment:
Do we ever pass any --feature flags when trimming the libraries where this might kick in? Should the optimization check go above this loop?